From 845158cf536f6a7f9a023585ef923d97f016ac59 Mon Sep 17 00:00:00 2001
From: "Sean P. McDonald" <sean.mcdonald@puppetlabs.com>
Date: Thu, 2 Aug 2018 15:49:46 -0700
Subject: [PATCH] turn down optimization on aarch
#     OpenSSL is failing to operate on aarch64 linux when built in a standard
#     format. Per suggestion of OpenSSL themselves
#     https://github.com/openssl/openssl/blob/OpenSSL_1_1_0h/INSTALL#L678L680
#     this commit patches the ssl configuration to use -O2 instead of -O3 when
#     compiling the tool. This seems to have fixed the issue
#                           - Sean P. McDonald 8/3/18

---
 Configurations/10-main.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index b49f04b5d7..3f61b7c318 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -633,7 +633,7 @@ sub vms_info {
         cc               => "gcc",
         cflags           => combine(picker(default => "-Wall",
                                            debug   => "-O0 -g",
-                                           release => "-O3"),
+                                           release => "-O2"),
                                     threads("-pthread")),
         ex_libs          => add("-ldl", threads("-pthread")),
         bn_ops           => "BN_LLONG RC4_CHAR",
--
2.18.0.windows.1

